breadbox/breadbox/crud/dataset.py

```diff
@@ -1650,11 +1677,30 @@ def get_subsetted_tabular_dataset_df(
 
     # Need to index by "value" after checking if empty db bc empty db has no 'value' keyword
     subsetted_tabular_dataset_df = pivot_df["value"]
-    # TODO: It seems like None data values are potentially stored as 'nan' in the db. Must fix this!
-    subsetted_tabular_dataset_df = subsetted_tabular_dataset_df.replace({np.nan: None})
+    # set typing for columns
+    col_dtypes = _get_column_types(
+        dataset.columns_metadata, tabular_dimensions_info.columns
+    )
+    subsetted_tabular_dataset_df = _convert_subsetted_tabular_df_dtypes(
+        subsetted_tabular_dataset_df, col_dtypes
+    )
     return subsetted_tabular_dataset_df
 
 
+def _convert_subsetted_tabular_df_dtypes(df: pd.DataFrame, dtype_map: Dict[str, Any]):
+    # Replace string boolean values with boolean
+    bool_cols_false = {}
+    bool_cols_true = {}
+    for col, dtype in dtype_map.items():
```

Comment:
Alternatively, what do you think of either converting it column by column?

Right now, it's a bit of a special case for boolean columns and it took me quiet a bit of thinking to understand what was going on. I think part of my delay in following is due to how much types shape my thinking.

The lines:
```
converted_df = df.replace(bool_cols_false, False)
converted_df = converted_df.replace(bool_cols_true, True)
```
Seemed strange because between the columns are going from string columns to a mix of booleans and strings, to finally all booleans, but the dtype is probably still "object". It's not until the very end the column type changes to bool.

Something like:

```
for col, dtype in dtype_map.items():
   column = df[col]
   if dtype == pd.BooleanDtype():
        column = column.replace({"True": True, "False": False})
   column = column.astype(dtype)
   df[col] = column
```

I think having a loop where each column is converted would making it more obvious. I admit this is subjective, but personally would help me.